source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1825,6 +1825,18 @@ ClusterManagerPtr ProdClusterManagerFactory::clusterManagerFromProto(
       http_context_, grpc_context_, router_context_, server_)};
 }
 
+absl::optional<Http::HttpServerPropertiesCache::Origin>
+getOrigin(const Network::TransportSocketOptionsConstSharedPtr& options, HostConstSharedPtr host) {
+  std::string sni(host->transportSocketFactory().defaultServerNameIndication());
+  if (options && options->serverNameOverride().has_value()) {
+    sni = options->serverNameOverride().value();
+  }
+  if (sni.empty()) {
+    return absl::nullopt;
```

Comment:
I'm slightly surprised that this only returns origins for https connections, not for http connections too. Is that just because we don't need it?

In any case, it looks like this function should probably live in an anon namespace to ensure that it has internal linkage.

Reply:
yeah no use case yet, we hard code https, and moved.